hedera-node/docs/design/services/token-service/airdrops/max-auto-associations-changes.md

```diff
@@ -0,0 +1,123 @@
+# Adapt existing logic with the new maxAutoAssociations behaviour
+
+## Purpose
+
+One of the main points of friction in Hedera regarding token airdrops is the need for a receiver to associate with a token before the sender of the token attempts the transfer. Currently a way to reduce this friction is to use the ability of accounts to automatically associate with tokens. This is controlled by their `maxAutoAssociations` property which is the number of automatic token associations the account can support (i.e. slots). When an account is created or the property value is updated the account prepays for all of the `maxAutoAssociations`.
+
+We can reduce this friction by further expanding on the functionality around auto associations by having the ability to have unlimited number of auto association slots and removing the requirement for an account to prepay for unused token association slots. Instead we can have the sender pay for the initial number of auto association slots needed for a transfer.
+
+Additionally, in order to support behaviour similar to other EVM-compatible chains, we will want to initialize auto- and lazy-created accounts to have unlimited number of auto association slots.
+
+In order to implement [HIP-904](https://hips.hedera.com/hip/hip-904) we need to adapt part of the existing Services logic, so that we are aligned with the new approach.
+
+## Goals
+
+1. Implement the ability for accounts to have unlimited number of auto associations
+2. Accounts shouldn’t prepay for unused token auto associations slots. Accounts should pay for slots only when they are used
+3. Senders should cover the initial fees related to auto association
+4. Initialize auto- and lazy-created accounts to have unlimited number of auto associations
+
+## Non-Goals
+
+- Updating existing accounts’ `maxAutoAssociations` values
+
+## Architecture
+
+The implementation related to the new `maxAutoAssociations` behaviour will be gated behind a feature flag.
```

Comment:
Let's give the feature flag a name here

Reply:
Added as `tokens.airdrops.maxAutoAssociations.enabled`, let me know if some different naming makes better sense.

Reply:
Updated to `tokens.airdrops.unlimitedMaxAutoAssociations.enabled` as it seemed more clear.